pkgs/applications/virtualization/virt-manager/default.nix

```diff
@@ -25,7 +26,7 @@ python3Packages.buildPythonApplication rec {
 
   buildInputs = [
     libvirt-glib vte dconf gtk-vnc gnome3.adwaita-icon-theme avahi
-    gsettings-desktop-schemas libosinfo gtk3
+    gsettings-desktop-schemas libosinfo gtk3 gtksourceview4 librsvg
```

Comment:
Actually yes, it would make more sense to move `wrapGAppsHook` to `buildInputs` as it already propagates `librsvg`.
The problem is that virtmanager tries to update icon cache in the `setup.py`.
I'm not sure how to pass a flag to disable this behavior to `self.distribution` in the `setup.py`.

I can move `gtk3` to `nativeBuildInputs` to make it work, but first want to try doing this via the `setup.py` flag.

Reply:
All you need to do here is
```
setupPyBuildFlags = [
  "--no-update-icon-cache"
];
```

We also should be able to drop the bit about compiling schemas below since that's done in `setup.py`.

Reply:
Unfortunately that would not fork as it passes flags to `build_ext` while virtmanager's `setup.py` checks `self.distribution`.

I've managed to make it work via custom `buildPhase`:
```
  buildPhase = ''
    runHook preBuild
    cp ${setuppy} nix_run_setup
    ${python3Packages.python.pythonForBuild.interpreter} nix_run_setup --no-update-icon-cache build_ext bdist_wheel
    runHook postBuild
  '';
```

Maybe we need to add a new `setupPyDistFlags` and modify build phases to include
```
... nix_run_setup ${(lib.concatStringsSep " " setupPyBuildFlags)} ... build_ext ...
```

@FRidh What do you think about this? I can make a PR if it's an acceptable change. Would incur mass rebuild though.

Reply:
Sounds good to me @aske, I noticed that shortly after commenting this.